getting_started/workflow/best_practices/what_are_godot_classes.rst

```diff
@@ -1,127 +1,66 @@
 
-.. note::
+The engine offers two main ways to create reusable objects: scripts and scenes. Neither of these
+technically defines classes under the hood.
 
-   As a result, you can instance scripts without the ``extends`` keyword
-   from code, but you cannot attach them to a :ref:`Node <class_Node>`
+Still, many best practices using Godot involve applying object-oriented programming principles to
+the scripts and scenes that compose your game. That is why it's useful to understand how we can
+think of them as classes.
 
+This guide briefly explains how scripts and scenes work in the engine's core to help you understand
+how scripts and scenes work under the hood.
 
-Scripting performances and PackedScene
---------------------------------------
+How scripts work in the engine
+------------------------------
 
-As the size of Objects increases, the scripts' necessary size to create them
-grows much, much larger. Creating node hierarchies demonstrates this. Each
-individual Node's logic could be several hundred lines of code in length.
+The engine provides built-in classes like :ref:`Node <class_Node>`. You can extend those to create
+derived types using a script.
 
-Let's see a simple example of creating a single ``Node`` as a child. The code
-below creates a new ``Node``, changes its name, assigns a script to it, sets its
-future parent as its owner so it gets saved to disk along with it, and finally
-adds it as a child of the ``Main`` node:
+These scripts are not technically classes. Instead, they are resources that tell the engine a
+sequence of initializations to perform on one of the engine's built-in classes.
 
-.. tabs::
-  .. code-tab:: gdscript GDScript
+Godot's internal classes have methods that register a class's data with a :ref:`ClassDB
+<class_ClassDB>`. This database provides runtime access to class information. ``ClassDB`` contains
+information about classes like:
 
-    # Main.gd
-    extends Node
+- Properties.
+- Methods.
+- Constants.
+- Signals.
 
-    func _init():
-        var child = Node.new()
-        child.name = "Child"
-        child.script = preload("Child.gd")
-        child.owner = self
-        add_child(child)
+This ``ClassDB`` is what objects check against when performing an operation like accessing a
+property or calling a method. It checks the database's records and the object's base types' records
+to see if the object supports the operation.
 
-  .. code-tab:: csharp
+Attaching a :ref:`Script <class_Script>` to your object extends the methods, properties, and signals
+available from the ``ClassDB``.
 
-    using System;
-    using Godot;
+.. note::
 
-    namespace ExampleProject
-    {
-        public class Main : Resource
-        {
-            public Node Child { get; set; }
+    Objects check their attached script before the database, which is why scripts can override
+    built-in methods. If a script defines a ``_get_property_list()`` method, Godot merges that data
```

Comment:
I think explaining how `_get_property_list()` works is too out-of-scope for an article about applying object-oriented principles. It's also something very rarely used in projects, I've never used it, and I just did a recursive grep for all the projects on my computer and the only instance is in a GDNative project.

Reply:
Well, the specific subsection within the article is talking about the relationship between a Script and its contributions to the reflection data collected from ClassDB, so it's *kinda* on topic in the sense that it describes further how Scripts contribute to class information.